service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/EncounterExtractor.java

```diff
@@ -0,0 +1,52 @@
+package uk.nhs.adaptors.gp2gp.ehr.utils;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.Encounter;
+import org.hl7.fhir.dstu3.model.ListResource;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.fhir.dstu3.model.ResourceType;
+
+public final class EncounterExtractor {
+    private static final String CONSULTATION_LIST_CODE = "1149501000000101";
+
+    private EncounterExtractor() {
```

Comment:
Is this constructor needed?

Reply:
It prevents creating an instance of this class. It is for static use only.

Reply:
right, I didn't notice it is private, what about @NoArgsConstructor(access = AccessLevel.PRIVATE) from lombok ?